library/Closed/Internal.hs

```diff
@@ -164,25 +164,25 @@ unrepresentable x cx prefix =
 natToClosed :: (n <= x, x <= m, KnownNat x, KnownNat n, KnownNat m) => proxy x -> Closed n m
 natToClosed p = Closed $ natVal p
 
--- | Add one inhabitant at the end
-weakenUpper :: Closed n m -> Closed n (m + 1)
+-- | Add inhabitants at the end
+weakenUpper :: (n <= m, m <= k) => Closed n m -> Closed n k
```

Comment:
You could add an explicit `forall` here for type application goodness
```
weakenUpper @5 (x :: Closed 0 3)
```

Reply:
Done, good call

Reply:
Just call me the "`TypeApplications` whisperer"